### PR TITLE
use fused batch norm and make gamma optional

### PR DESCRIPTION
**Pull request by AndreasMadsen**
Fused batch norm is recommended in the [tensorflow/performance_guide](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/docs_src/performance/performance_guide.md). I couldn't use `tf.contrib.layers.batch_norm` because it creates its own variables and I couldn't get `reuse=True` to work.

It also makes gamma optional since that is redundant when the activation function is multiplicative linear (e.g. ReLU in ByteNet).

I see approximately a 20% better performance in ByteNet when using these optimizations.

_#19 can be closed when this is merged._

### CODE REVIEW

**Comment by buriburisuri** [comment]
@AndreasMadsen Wow~ this is great contribution! Thank you very much.
